src/postamble.js

```diff
@@ -131,11 +131,10 @@ function stackCheckInit() {
 }
 #endif
 
-#if RELOCATABLE
-var dylibsLoaded = false;
-#if '$LDSO' in addedLibraryItems
-LDSO.init();
-#endif
+#if MAIN_MODULE && PTHREADS
+// Map of modules to be shared with new threads.  This gets populated by the
+// main thread an shared with all new workers.
+var sharedModules = Module['sharedModules'] ||[];
```

Comment:
Does this really happen before `Module['sharedModules']` is set? But this is at the top level scope, so it happens immediately, while the shared modules arrive in a message..?

Reply:
Yes that way this works in worker.js is that we set a bunch of properties on the Worker object before importing the main script.  e.g.: https://github.com/emscripten-core/emscripten/blob/32043824e01ec6154b700bc410f30473fb277732/src/worker.js#L177

From the POV of the main script these act like incoming properties of the Module, just like if they had been set by a user in the non-threaded case.

Reply:
Ah, thanks, I forgot that stuff...